### ISSUE TITLE
Cannot assign a new value to a local variable in release android (regression in 3.1 beta 3)

### ISSUE THREAD

**Issue by DrMoriarty** [godotengine/godot#25408]
**Godot version:**

Godot 3.1.beta3 (commit 953cd03ea)

**OS/device including version:**

MacOSX 10.13.6, Android 7.1.2

**Issue description:**

I rebuilt godot and android templates. When I run a game in debug mode, it works pretty well.
When I run release build it crashes with this error:
```
01-28 12:49:36.884: I/godot(19244): Loading module: org/godotengine/godot/GodotFacebook
01-28 12:49:36.888: W/GodotView(19244): creating OpenGL ES 3.0 context :
01-28 12:49:36.924: I/godot(19244): OpenGL ES 3.0 Renderer: Adreno (TM) 506
01-28 12:49:37.507: E/godot(19244): **SCRIPT ERROR**: Parse Error: Cannot assign a new value to a constant.
01-28 12:49:37.507: E/godot(19244):    At: res://common/config.gde:495:GDScript::load_byte_code() - Parse Error: Cannot assign a new value to a constant.
01-28 12:49:37.507: E/godot(19244): **ERROR**: Method/Function Failed, returning: ERR_PARSE_ERROR
01-28 12:49:37.507: E/godot(19244):    At: modules/gdscript/gdscript.cpp:784:load_byte_code() - Method/Function Failed, returning: ERR_PARSE_ERROR
01-28 12:49:37.510: E/godot(19244): **ERROR**: Condition ' err != OK ' is true. returned: RES()
01-28 12:49:37.510: E/godot(19244):    At: modules/gdscript/gdscript.cpp:2148:load() - Condition ' err != OK ' is true. returned: RES()
01-28 12:49:37.510: E/godot(19244): **ERROR**: Method/Function Failed, returning: RES()
01-28 12:49:37.510: E/godot(19244):    At: core/io/resource_loader.cpp:279:_load() - Method/Function Failed, returning: RES()
01-28 12:49:37.510: E/godot(19244): **ERROR**: Condition ' res.is_null() ' is true. Continuing..:
01-28 12:49:37.510: E/godot(19244):    At: main/main.cpp:1540:start() - Condition ' res.is_null() ' is true. Continuing..:
```
and a little bit later:
```
01-28 12:49:47.006: A/libc(19244): Fatal signal 11 (SIGSEGV), code 1, fault addr 0x8 in tid 19310 (GLThread 45643)
01-28 12:49:47.006: A/libc(19244): [ 01-28 12:49:47.009 13850:13850 W/         ]
01-28 12:49:47.006: A/libc(19244): debuggerd: handling request: pid=19244 uid=10479 gid=10479 tid=19310
01-28 12:49:47.006: A/libc(19244): [ 01-28 12:49:47.010 13850:13850 E/         ]
01-28 12:49:47.006: A/libc(19244): debuggerd: waitpid returned unexpected pid (0), committing murder-suicide
01-28 12:49:47.006: A/libc(19244): [ 01-28 12:49:47.010 13850:13850 W/         ]
01-28 12:49:47.006: A/libc(19244): debuggerd: pid 0 received signal 9
01-28 12:49:47.006: A/libc(19244): [ 01-28 12:49:47.010 13850:13850 E/         ]
01-28 12:49:47.006: A/libc(19244): debuggerd: killing target 19244
01-28 12:49:47.006: A/libc(19244): [ 01-28 12:49:47.050 19371:19371 I/         ]
01-28 12:49:47.006: A/libc(19244): debuggerd: starting
```

I checked my script file. In config.gd string 495 is a regular assign to a local variable. It not used enums or something complicated things.
```
 486|            Log.info('Load param: %s %s'%[k, param.Value])
 487|            var changed = false
 488|            var re = RegEx.new()
 489|            re.compile('unlocked_level_(\\d+)')
 490|            if re.find(k) >= 0:
 491|                var ln = int(re.get_capture(1))
 492|                if ln >= 200:
 493|                    ln = 200-ln
 494|                    k = 'unlocked_level_%d'%ln
 495|                    changed = true                 <------ it blames this line
 496|            var re2 = RegEx.new()
 497|            re2.compile('level_stars_(\\d+)')
 498|            if re2.find(k) >= 0:
 499|                var ln = int(re2.get_capture(1))
 500|                if ln >= 200:
 501|                    ln = 200-ln
 502|                    k = 'level_stars_%d'%ln
 503|                    changed = true
```

I compiled my game in release mode on previous beta and it also works pretty well. I didn't changed this file since prev build.
**Steps to reproduce:**

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
Likely a regression from one of those commits: #25274

**Comment by hedin-hiervard** [comment]
It happens to me too.

**Comment by mikeKulasinski** [comment]
Not sure if it is related but on master branch from today I can't run the app in debug mode, looks like a problem with resource resolving

> 019-02-21 22:22:57.654 15845-15845/? I/: Late-enabling -Xcheck:jni
2019-02-21 22:22:57.937 15845-15875/ D/libEGL: loaded /vendor/lib64/egl/libEGL_adreno.so
2019-02-21 22:22:57.943 15845-15875/ E/libc: Access denied finding property "vendor.debug.egl.profiler"
2019-02-21 22:22:57.940 15845-15845/ W/: type=1400 audit(0.0:77): avc: denied { read } for comm=45474C20496E6974 name="u:object_r:vendor_default_prop:s0" dev="tmpfs" ino=16298 scontext=u:r:untrusted_app:s0:c144,c257,c512,c768 tcontext=u:object_r:vendor_default_prop:s0 tclass=file permissive=0
2019-02-21 22:22:57.945 15845-15875/ D/libEGL: loaded /vendor/lib64/egl/libGLESv1_CM_adreno.so
2019-02-21 22:22:57.958 15845-15875/ D/libEGL: loaded /vendor/lib64/egl/libGLESv2_adreno.so
2019-02-21 22:22:58.034 15845-15845/ D/OpenGLRenderer: Skia GL Pipeline
2019-02-21 22:22:58.049 15845-15877/ I/godot: Android modules: Nothing to load, aborting
2019-02-21 22:22:58.076 15845-15879/ I/Adreno: QUALCOMM build                   : 6fb5a5b, Ife855c4895
2019-02-21 22:22:58.076 15845-15879/ I/Adreno: Build Config                     : S L 4.0.10 AArch64
2019-02-21 22:22:58.079 15845-15879/ I/Adreno: PFP: 0x005ff110, ME: 0x005ff066
2019-02-21 22:22:58.082 15845-15879/ I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 1
2019-02-21 22:22:58.082 15845-15879/ I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2019-02-21 22:22:58.083 15845-15879/ I/OpenGLRenderer: Initialized EGL, version 1.4
2019-02-21 22:22:58.083 15845-15879/ D/OpenGLRenderer: Swap behavior 1
2019-02-21 22:22:58.089 15845-15877/ W/GodotView: creating OpenGL ES 3.0 context :
2019-02-21 22:22:58.094 15845-15877/ I/godot: OpenGL ES 3.0 Renderer: Adreno (TM) 540
2019-02-21 22:22:58.123 15845-15877/ I/libOpenSLES: Emulating old channel mask behavior (ignoring positional mask 0x3, using default mask 0x3 based on channel count of 2)
2019-02-21 22:22:58.126 15845-15877/ W/AudioTrack: AUDIO_OUTPUT_FLAG_FAST denied by server; frameCount 0 -> 1772
2019-02-21 22:22:58.359 15845-15877/ E/godot: **SCRIPT ERROR**: Parse Error: Expected '(' after identifier (syntax: 'func <identifier>([arguments]):' ).
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: res://GameBoard.gdc:2:GDScript::load_byte_code() - Parse Error: Expected '(' after identifier (syntax: 'func <identifier>([arguments]):' ).
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Method/Function Failed, returning: ERR_PARSE_ERROR
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: modules/gdscript/gdscript.cpp:778:load_byte_code() - Method/Function Failed, returning: ERR_PARSE_ERROR
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: modules/gdscript/gdscript.cpp:2142:load() - Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Failed loading resource: res://GameBoard.gdc
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: core/io/resource_loader.cpp:279:_load() - Method/Function Failed, returning: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: res://GameBoard.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://GameBoard.gd
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: scene/resources/scene_format_text.cpp:439:poll() - res://GameBoard.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://GameBoard.gd
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: core/io/resource_loader.cpp:202:load() - Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Failed loading resource: res://GameBoard.tscn
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: core/io/resource_loader.cpp:279:_load() - Method/Function Failed, returning: RES()
2019-02-21 22:22:58.359 15845-15877/ E/godot: **ERROR**: Failed loading scene: res://GameBoard.tscn
2019-02-21 22:22:58.359 15845-15877/ E/godot:    At: main/main.cpp:1727:start() - Condition ' !scene ' is true. returned: false
2019-02-21 22:22:58.458 15845-15877/ E/godot: **SCRIPT ERROR**: Parse Error: Expected '(' after identifier (syntax: 'func <identifier>([arguments]):' ).
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: res://GameBoard.gdc:2:GDScript::load_byte_code() - Parse Error: Expected '(' after identifier (syntax: 'func <identifier>([arguments]):' ).
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: Method/Function Failed, returning: ERR_PARSE_ERROR
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: modules/gdscript/gdscript.cpp:778:load_byte_code() - Method/Function Failed, returning: ERR_PARSE_ERROR
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: modules/gdscript/gdscript.cpp:2142:load() - Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: Failed loading resource: res://GameBoard.gdc
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: core/io/resource_loader.cpp:279:_load() - Method/Function Failed, returning: RES()
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: res://GameBoard.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://GameBoard.gd
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: scene/resources/scene_format_text.cpp:439:poll() - res://GameBoard.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://GameBoard.gd
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.458 15845-15877/ E/godot:    At: core/io/resource_loader.cpp:202:load() - Condition ' err != OK ' is true. returned: RES()
2019-02-21 22:22:58.458 15845-15877/ E/godot: **ERROR**: Failed loading resource: res://GameBoard.tscn

**Comment by hedin-hiervard** [comment]
Still happens to me, beta6, OS X release build

**Comment by akien-mga** [comment]
Related to #25334.

**Comment by akien-mga** [comment]
Should be fixed by #26521.
